gui/layers/_markers.py

```diff
@@ -0,0 +1,258 @@
+from typing import Union
+
+import numpy as np
+
+from ._base import Layer
+from ._register import add_to_viewer
+from .._vispy.scene.visuals import Markers as MarkersNode
+
+
+@add_to_viewer
+class Markers(Layer):
+    """Markers layer.
+
+    Parameters
+    ----------
+    marker_coords : np.ndarray
+        coordinates for each marker.
+
+    symbol : str
+        symbol to be used as a marker
+
+    size : int, float, np.ndarray, list
+        size of the marker. If given as a scalar, all markers are the
+        same size. If given as a list/array, size must be the same
+        length as marker_coords and sets the marker size for each marker
+        in marker_coords (element-wise).
+
+    edge_width : float, None
+        width of the symbol edge in px
+            vispy docs say: "exactly one edge_width and
+            edge_width_rel must be supplied"
+
+    edge_width_rel : float, None
+        width of the marker edge as a fraction of the marker size.
+
+            vispy docs say: "exactly one edge_width and
+            edge_width_rel must be supplied"
+
+    edge_color : Color, ColorArray
+        color of the marker border
+
+    face_color : Color, ColorArray
+        color of the marker body
+
+    scaling : bool
+        if True, marker rescales when zooming
+
+    See vispy's marker visual docs for more details:
+    http://api.vispy.org/en/latest/visuals.html#vispy.visuals.MarkersVisual
+
+
+
+    """
+
+    def __init__(
+        self, marker_coords, symbol='o', size=10, edge_width=1,
+            edge_width_rel=None, edge_color='black', face_color='white',
+            scaling=True):
+
+        visual = MarkersNode()
+        super().__init__(visual)
+
+        # Save the marker coordinates
+        self._marker_coords = marker_coords
+
+        # Save the marker style params
+        self._symbol = symbol
+        self._size = size
+        self._edge_width = edge_width
+        self._edge_width_rel = edge_width_rel
+        self._edge_color = edge_color
+        self._face_color = face_color
+        self._scaling = scaling
+
+        # update flags
+        self._need_display_update = False
+        self._need_visual_update = False
+
+    @property
+    def marker_coords(self) -> np.ndarray:
+        """ndarray: coordinates of the marker centroids
+        """
+        return self._marker_coords
+
+    @property
+    def data(self) -> np.ndarray:
+        """ndarray: coordinates of the marker centroids
+        """
+        return self._marker_coords
+
+    @data.setter
+    def data(self, data: np.ndarray) -> None:
+        self._marker_coords = data
+
+        self.viewer._child_layer_changed = True
```

Comment:
I think that there should be a setter for `marker_coords` that performs this instead and that `data` really just wraps all of that. Or maybe conversely we can just use `data` in place of `marker_coords` entirely.

Reply:
Good point. I like having marker_coords() instead of just using data because it's more clear what the "data" are. Based on the base Layer class, it seems that `data` is a required properties. Thus, I think it's best to have a setter method for `marker_coords` and then call it in the `data` setter method. What do you think?

Reply:
Honestly, either solution is fine with me. My only comment is that calling it `marker_coords` instead of just `coords` seems a bit redundant, given that it is a “Marker” layer and we have the possibility to shorten the descriptor.

@sofroniewn thoughts?

Reply:
`coords` or `coordinates` seems better to me